src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs

```diff
@@ -55,6 +56,15 @@ public class RestoreResult : IRestoreResult
         /// </summary>
         public TimeSpan ElapsedTime { get; }
 
+        /// <summary>
+        ///  Cache File. The previous cache file for this project
+        /// </summary>
+        public CacheFile CacheFile { get; } 
+        /// <summary>
+        /// Cache File path. The file path where the cache is written out
+        /// </summary>
+        public string CacheFilePath { get;  }
```

Comment:
I don't see any consumers of CacheFile or CacheFilePath. It could be useful in the future, but for now I suggest removing these public properties and keeping the cache file as an internal implementation detail. Callers should trust that restore does the right thing in terms of caching, and that the noop flag is followed.